src/helpers/oracleClient.ts

```diff
@@ -2,18 +2,47 @@ import { oracle } from "@uma/sdk";
 import config from "constants/oracleConfig";
 import Events from "events";
 
+type OracleType = oracle.types.state.OracleType;
+type State = oracle.types.state.State;
+type Client = oracle.client.Client;
+
+export const isSupportedOracleType = oracle.utils.isSupportedOracleType;
+
 export { oracle };
+
+export type { OracleType, State, Client };
 export const events = new Events();
-export const client = oracle.client.factory(
-  config,
-  (state: oracle.types.state.State) => events.emit("change", state)
+export const clients = oracle.factory(
+  // the configs are an array of oracle configs, the order of the configs determine the order of clients returned
+  Object.fromEntries(config),
+  (oracleType: OracleType, state: State) =>
+    events.emit("change", oracleType, state)
 );
 
-// setting this to the fastest interval for more responsive state changes
-client.startInterval(1);
+// Iterate easily over all clients, useful for duplicating commands on clients
+export function forEach(cb: (client: Client) => void): void {
+  Object.values(clients).forEach(cb);
+}
+
+export function mapObject<V>(cb: (client: Client) => V): {
+  [key in OracleType]?: V;
```

Comment:
Hmmm, unfamiliar with this syntax. What does the ? do when it's a mapping? Aren't mapping types optional anyway, since they don't expect _all_ keys to be defined?

Reply:
when mapping an enum it seems to expect all enum keys to be defined, i think this says that not all enums will be defined